pandas/tseries/offsets.py

```diff
@@ -507,8 +513,18 @@ def freqstr(self):
         else:
             fstr = code
 
+        try:
+            if self._offset:
```

Comment:
is this still needed?

Reply:
Yes.  Not all subclasses define the `_offset` attribute.  That is something I intend to standardize, but this PR is explicitly intended to be limited in scope.